kubectl-direct-csi/util/util.go

```diff
@@ -0,0 +1,750 @@
+
+	// Create StorageClass for the new driver
+	storageClass := &storagev1.StorageClass{
+		metav1.TypeMeta{
+			Kind:       "StorageClass",
+			APIVersion: "storage.k8s.io/v1",
+		},
+		metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				createByLabel: directCSIController,
+			},
+		},
+		name, // Provisioner
+		parameters,
+		&retainPolicy,
+		mountOptions,
+		&allowExpansion,
+		&bindingMode,
+		allowedTopologies,
+	}
+
+	return retry(func() error {
+		_, err := kClient.StorageV1().StorageClasses().Create(ctx, storageClass, metav1.CreateOptions{})
+		return err
+	})
+}
+
+func CreateRBACRoles(ctx context.Context, kClient clientset.Interface, name string) error {
+	if err := createServiceAccount(ctx, kClient, name); err != nil {
+		return err
+	}
+	if err := createClusterRole(ctx, kClient, name); err != nil {
+		return err
+	}
+	if err := createClusterRoleBinding(ctx, kClient, name); err != nil {
+		return err
+	}
+	return nil
+}
+
+func createServiceAccount(ctx context.Context, kClient clientset.Interface, name string) error {
+	serviceAccount := &corev1.ServiceAccount{
+		metav1.TypeMeta{
+			Kind:       "ServiceAccount",
+			APIVersion: "v1",
+		},
+		metav1.ObjectMeta{
+			Name:      name,
+			Namespace: name,
+			Labels: map[string]string{
+				createByLabel: directCSIController,
+			},
+		},
+		[]corev1.ObjectReference{},
+		[]corev1.LocalObjectReference{},
+		nil,
+	}
+
+	return retry(func() error {
+		_, err := kClient.CoreV1().ServiceAccounts(name).Create(ctx, serviceAccount, metav1.CreateOptions{})
+		return err
+	})
+}
+
+func createClusterRoleBinding(ctx context.Context, kClient clientset.Interface, name string) error {
+	clusterRoleBinding := &rbacv1.ClusterRoleBinding{
+		metav1.TypeMeta{
+			Kind:       "ClusterRoleBinding",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				createByLabel: directCSIController,
+			},
+			Annotations: map[string]string{
+				"rbac.authorization.kubernetes.io/autoupdate": "true",
+			},
+		},
+		[]rbacv1.Subject{
+			rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				Name:      name,
+				Namespace: name,
+			},
+		},
+		rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     name,
+		},
+	}
+
+	return retry(func() error {
+		_, err := kClient.RbacV1().ClusterRoleBindings().Create(ctx, clusterRoleBinding, metav1.CreateOptions{})
+		return err
+	})
+}
+
+func createClusterRole(ctx context.Context, kClient clientset.Interface, name string) error {
+	clusterRole := &rbacv1.ClusterRole{
+		metav1.TypeMeta{
+			Kind:       "ClusterRole",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				createByLabel: directCSIController,
+			},
+			Annotations: map[string]string{
+				"rbac.authorization.kubernetes.io/autoupdate": "true",
+			},
+		},
+		[]rbacv1.PolicyRule{
+			rbacv1.PolicyRule{
```

Comment:
There is a new RBAC role I added to create CRDs. Please update this to add that RBAC role